admin/calendar_ics.php

```diff
@@ -22,73 +30,86 @@
         if (check_allow('login') && $expadmindata['disabled']!='y'
             && (check_allow('calendar_export_my') || check_allow('calendar_export_all'))
         ) {
-            if ($all==true && !check_allow('calendar_export_all')) $all=false;
-        } else { $continue=false; $message="no rights to export"; }
-    } else { $continue=false; $message="invalid token"; }
+            if ($all==true && !check_allow('calendar_export_all')) {
+                $all=false;
+            }
+        } else { 
+            $continue=false; 
+            $message="no rights to export"; 
+        }
+    } else { 
+        $continue=false; 
+        $message="invalid token"; 
+    }
 }
 
+
 if ($continue) {
+    $labs=laboratories__get_laboratories();
+    $laboratory_id=false;
+    if(isset($_REQUEST['lab_id']) && $_REQUEST['lab_id']) {
+        if(isset($labs[$_REQUEST['lab_id']])) {
+            $laboratory_id=$_REQUEST['lab_id'];
+        }
+    }
 
-$displayfrom_lower = time()-60*60*24*31*$settings['calendar_export_months_back'];
-$displayfrom_upper = time()+60*60*24*31*$settings['calendar_export_months_ahead'];
+    $displayfrom_lower = time()-60*60*24*31*$settings['calendar_export_months_back'];
+    $displayfrom_upper = time()+60*60*24*31*$settings['calendar_export_months_ahead'];
 
-if ($all) $expadminid=false; else $expadminid=$expadmindata['admin_id'];
-$results = calendar__get_events(true, $displayfrom_lower, $displayfrom_upper,$expadminid);
+    if ($all) $expadminid=false; else $expadminid=$expadmindata['admin_id'];
```

Comment:
If you are changing this line anyway, might worth rewriting if/else statement with curly braces.